lib/stdlib/src/rand.erl

```diff
@@ -81,27 +83,29 @@ export_seed_s({#{type:=Alg}, Seed}) -> {Alg, Seed}.
 %% seed({Alg,Seed}) setup RNG with a previously exported seed
 %% and return the NEW state
 
--spec seed(AlgOrExpState::alg() | export_state()) -> state().
+-spec seed(AlgOrStateOrExpState::builtin_alg() | state() | export_state()) -> state().
 seed(Alg) ->
     seed_put(seed_s(Alg)).
 
--spec seed_s(AlgOrExpState::alg() | export_state()) -> state().
+-spec seed_s(AlgOrStateOrExpState::builtin_alg() | state() | export_state()) -> state().
 seed_s(Alg) when is_atom(Alg) ->
```

Comment:
To minimize the use of guards maybe reorder these into:
```erlang
seed_s({AlgHandler,_Seed}) when is_map(AlgHandler) ->
seed_s({Alg0,Seed}) ->
seed_s(Alg) ->```
Then we rely on `alg_handler()` being a map and does not use that `alg()` must be an atom.